src/NSwag.SwaggerGeneration.WebApi/Processors/SwaggerResponseBuilder.cs

```diff
@@ -0,0 +1,218 @@
+
+                OperationResponseModels.Add(new OperationResponseModel(httpStatusCode, returnType, isNullable, description));
+            }
+        }
+
+        public void PopulateModelsFromProducesAttributes(IEnumerable<Attribute> producesResponseTypeAttributes)
+        {
+            foreach (dynamic producesResponseTypeAttribute in producesResponseTypeAttributes)
+            {
+                var returnType = producesResponseTypeAttribute.Type;
+                var httpStatusCode = producesResponseTypeAttribute.StatusCode.ToString(CultureInfo.InvariantCulture);
+                var description = HttpUtilities.IsSuccessStatusCode(httpStatusCode) ? _successXmlDescription : string.Empty;
+                OperationResponseModels.Add(new OperationResponseModel(httpStatusCode, returnType, true, description));
+            }
+        }
+
+
+        public async Task BuildSwaggerResponseAsync(
+            ParameterInfo returnParameter)
+        {
+            
+            foreach (var statusCodeGroup in OperationResponseModels.GroupBy(r => r.HttpStatusCode))
+            {
+                var httpStatusCode = statusCodeGroup.Key;
+                var returnType = statusCodeGroup.Select(r => r.ResponseType).FindCommonBaseType();
+                var description = string.Join("\nor\n", statusCodeGroup.Select(r => r.Description));
+
+                var typeDescription = _settings.ReflectionService.GetDescription(
+                    returnType, returnParameter?.GetCustomAttributes(), _settings);
+
+                var response = new SwaggerResponse
+                {
+                    Description = description ?? string.Empty
+                };
+
+                if (IsVoidResponse(returnType) == false)
+                {
+                    response.IsNullableRaw = OperationResponseModels.Any(r => r.IsNullable) && typeDescription.IsNullable;
+                    response.ExpectedSchemas = await GenerateExpectedSchemasAsync(statusCodeGroup);
+                    response.Schema = await _context.SchemaGenerator
+                        .GenerateWithReferenceAndNullability<JsonSchema4>(
+                            returnType, null, typeDescription.IsNullable, _context.SchemaResolver)
+                        .ConfigureAwait(false);
+                }
+
+                _context.OperationDescription.Operation.Responses[httpStatusCode] = response;
+            }
+
+            bool loadDefaultSuccessResponseFromReturnType;
+            if (OperationResponseModels.Count > 0)
+            {
+                // If there are some attributes declared on the controller \ action, only return a default success response
+                // if a 2xx status code isn't already defined and the SwaggerDefaultResponseAttribute is declared.
+                var operationResponses = _context.OperationDescription.Operation.Responses;
+                var hasSuccessResponse = operationResponses.ContainsKey("200") || operationResponses.ContainsKey("204");
+
```

Comment:
Why is this check limited to `200/204`? instead of any 2XX?

Reply:
The current WebAPI generator had this and I left it as is - https://github.com/RSuter/NSwag/blob/master/src/NSwag.SwaggerGeneration.WebApi/Processors/OperationResponseProcessor.cs#L127. @RSuter?

Reply:
I think we should use 
https://github.com/RSuter/NSwag/blob/master/src/NSwag.Core/HttpUtilities.cs#L17
here. 

Please do such changes also in NSwag.SwaggerGeneration.WebApi

Reply:
This type is shared between the two - so changing it here should fix both.